### PR TITLE
#2066-doc-update-metrics-latest

### PR DESCRIPTION
**Pull request by nataliyah123**
Fixes part of #2066

### CODE REVIEW

File: tensorflow_addons/metrics/cohens_kappa.py [hunk 1]
```diff
@@ -38,28 +38,40 @@ class CohenKappa(Metric):
...
+    >>> metric = tfa.metrics.CohenKappa(num_classes=5, sparse_labels=True)
+    >>> metric.update_state(y_true = actuals, y_pred = preds)
+    <tf.Tensor: shape=(5, 5), dtype=float32, numpy=
+     array([[0., 0., 0., 0., 0.],
+            [0., 2., 0., 0., 0.],
+            [0., 0., 0., 0., 1.],
+            [0., 0., 0., 1., 0.],
+            [0., 0., 1., 0., 3.]], dtype=float32)>
+    >>> result = metric(y_true = actuals, y_pred = preds)
+    >>> print('Final result: ', result.numpy())
```
Comment (WindQAQ) [inline tensorflow_addons/metrics/cohens_kappa.py:53]: Can we use
```python
>>> result.numpy()
```
instead of
```python
print('Final result: ', result.numpy())
```
?

File: tensorflow_addons/metrics/cohens_kappa.py [hunk 2]
```diff
@@ -38,28 +38,40 @@ class CohenKappa(Metric):
...
+    >>> metric = tfa.metrics.CohenKappa(num_classes=5, sparse_labels=True)
+    >>> metric.update_state(y_true = actuals, y_pred = preds, sample_weight=weights)
+    <tf.Tensor: shape=(5, 5), dtype=float32, numpy=
+     array([[ 0.,  0.,  0.,  0.,  0.],
+            [ 0.,  6.,  0.,  0.,  0.],
+           [ 0.,  0.,  0.,  0., 10.],
+            [ 0.,  0.,  0.,  2.,  0.],
+            [ 0.,  0.,  2.,  0.,  7.]], dtype=float32)>
+    >>> result = metric(y_true = actuals, y_pred = preds)
+    >>> print('Final result: ', result.numpy())
```
Comment (WindQAQ) [inline tensorflow_addons/metrics/cohens_kappa.py:65]: ditto

File: tensorflow_addons/metrics/hamming.py [hunk 3]
```diff
@@ -38,14 +38,14 @@ def hamming_distance(actuals: TensorLike, predictions: TensorLike) -> tf.Tensor:
...
-                              dtype=tf.int32)
-    result = hamming_distance(actuals, predictions)
-    print('Hamming distance: ', result.numpy())
-    ```
+    >>> actuals = tf.constant([1, 1, 0, 0, 1, 0, 1, 0, 0, 1],
+    ... dtype=tf.int32)
+    >>> predictions = tf.constant([1, 0, 0, 0, 1, 0, 0, 1, 0, 1],
+    ... dtype=tf.int32)
+    >>> metric = hamming_distance(actuals, predictions)
+    >>> print('Hamming distance: ', metric.numpy())
```
Comment (WindQAQ) [inline tensorflow_addons/metrics/hamming.py:46]: ditto

File: tensorflow_addons/metrics/hamming.py [hunk 4]
```diff
@@ -84,31 +84,29 @@ def hamming_loss_fn(
...
+    >>> hl = HammingLoss(mode='multiclass', threshold=0.6)
+    >>> actuals = tf.constant([[1, 0, 0, 0],[0, 0, 1, 0],
+    ... [0, 0, 0, 1],[0, 1, 0, 0]], dtype=tf.float32)
+    >>> predictions = tf.constant([[0.8, 0.1, 0.1, 0],
+    ... [0.2, 0, 0.8, 0],[0.05, 0.05, 0.1, 0.8],[1, 0, 0, 0]],
+    ... dtype=tf.float32)
+    >>> hl.update_state(actuals,  predictions)
+    <tf.Variable 'UnreadVariable' shape=() dtype=float32, numpy=4.0>
+    >>> # uncomment the line below to see the result
+    >>> print('Hamming loss: ', hl.result().numpy())
```
Comment (WindQAQ) [inline tensorflow_addons/metrics/hamming.py:97]: ditto

File: tensorflow_addons/metrics/hamming.py [hunk 5]
```diff
@@ -84,31 +84,29 @@ def hamming_loss_fn(
...
+    Hamming loss: 0.25
+    >>> # multi-label hamming loss
+    >>> hl = HammingLoss(mode='multilabel', threshold=0.8)
+    >>> actuals = tf.constant([[1, 0, 1, 0],[0, 1, 0, 1],
+    ... [0, 0, 0,1]], dtype=tf.int32)
+    >>> predictions = tf.constant([[0.82, 0.5, 0.90, 0],
+    ... [0, 1, 0.4, 0.98],[0.89, 0.79, 0, 0.3]],dtype=tf.float32)
+    >>> hl.update_state(actuals, predictions)
+    <tf.Variable 'UnreadVariable' shape=() dtype=float32, numpy=3.0>
+    >>> print('Hamming loss: ', hl.result().numpy())
```
Comment (WindQAQ) [inline tensorflow_addons/metrics/hamming.py:107]: ditto

File: tensorflow_addons/metrics/matthews_correlation_coefficient.py [hunk 6]
```diff
@@ -42,18 +42,18 @@ class MatthewsCorrelationCoefficient(tf.keras.metrics.Metric):
...
+
+    >>> actuals = tf.constant([[1.0], [1.0], [1.0], [0.0]],
+    ... dtype=tf.float32)
+    >>> preds = tf.constant([[1.0], [0.0], [1.0], [1.0]],
+    ... dtype=tf.float32)
+    >>> # Matthews correlation coefficient
+    >>> metric = tfa.metrics.MatthewsCorrelationCoefficient(num_classes=1)
+    >>> metric.update_state(y_true = actuals, y_pred = preds)
+    >>> result = metric(y_true = actuals, y_pred = preds)
+    >>> print('Matthews correlation coefficient is :', result.numpy())
```
Comment (WindQAQ) [inline tensorflow_addons/metrics/matthews_correlation_coefficient.py:54]: ditto

File: tensorflow_addons/metrics/multilabel_confusion_matrix.py [hunk 7]
```diff
@@ -45,31 +45,43 @@ class MultiLabelConfusionMatrix(Metric):
...
+            [2., 0.]]], dtype=float32)
+    >>> # if multiclass input is provided
+    >>> y_true = tf.constant([[1, 0, 0], [0, 1, 0]],
+    ... dtype=tf.int32)
+    >>> y_pred = tf.constant([[1, 0, 0],[0, 0, 1]],
+    ... dtype=tf.int32)
+    >>> metric = tfa.metrics.MultiLabelConfusionMatrix(num_classes=3)
+    >>> metric.update_state(y_true, y_pred)
+    >>> result = metric(y_true, y_pred)
+    >>> print('Confusion matrix:', result.numpy()) #doctest: -DONT_ACCEPT_BLANKLINE
```
Comment (WindQAQ) [inline tensorflow_addons/metrics/multilabel_confusion_matrix.py:75]: ditto

File: tensorflow_addons/metrics/r_square.py [hunk 8]
```diff
@@ -60,13 +60,15 @@ class RSquare(Metric):
...
-     result.update_state(actuals, preds)
-     print('R^2 score is: ', r1.result().numpy()) # 0.57142866
-    ```
+
+    >>> actuals = tf.constant([1, 4, 3], dtype=tf.float32)
+    >>> preds = tf.constant([2, 4, 4], dtype=tf.float32)
+    >>> metric = tfa.metrics.r_square.RSquare()
+    >>> metric.update_state(y_true = actuals, y_pred = preds)
+    >>> result = metric(y_true = actuals,  y_pred = preds)
+    >>> print('R^2 score is: ', result.numpy())
```
Comment (WindQAQ) [inline tensorflow_addons/metrics/r_square.py:69]: ditto

File: tensorflow_addons/metrics/cohens_kappa.py [hunk 9]
```diff
@@ -38,28 +38,40 @@ class CohenKappa(Metric):
...
+    Final result:  0.42080373
 
     Usage with tf.keras API:
+    >>> inputs = tf.keras.Input(shape=(10,))
```
Comment (WindQAQ) [inline tensorflow_addons/metrics/cohens_kappa.py:69]: insert an empty line above

File: tensorflow_addons/metrics/hamming.py [hunk 10]
```diff
@@ -84,31 +84,29 @@ def hamming_loss_fn(
...
+    >>> # multi-class hamming loss
+    >>> hl = HammingLoss(mode='multiclass', threshold=0.6)
+    >>> actuals = tf.constant([[1, 0, 0, 0],[0, 0, 1, 0],
+    ... [0, 0, 0, 1],[0, 1, 0, 0]], dtype=tf.float32)
+    >>> predictions = tf.constant([[0.8, 0.1, 0.1, 0],
+    ... [0.2, 0, 0.8, 0],[0.05, 0.05, 0.1, 0.8],[1, 0, 0, 0]],
+    ... dtype=tf.float32)
+    >>> hl.update_state(actuals,  predictions)
+    <tf.Variable 'UnreadVariable' shape=() dtype=float32, numpy=4.0>
+    >>> # uncomment the line below to see the result
```
Comment (WindQAQ) [inline tensorflow_addons/metrics/hamming.py:96]: This line is meaningless in testable docstring. Can we remove it?

**Review by WindQAQ — COMMENTED** [review]
Thanks for the contribution! Some requests:

1. Can we canonicalize naming convention? Like
```python
>>> y_true = np.foo(xxx)
>>> y_pred = np.foo(xxx)
>>> metric = tfa.metrics.bar(yyy)
>>> metric.update_state(y_true, y_pred)
>>> result = metric.result()
>>> result.numpy()
```

2. Some comments are not so meaningful in testable docstring. Like `# metric name`: When users see the API page, they know this page is that metric. Therefore, we can also clean up this kind of comments in docstring in this PR!

File: tensorflow_addons/metrics/hamming.py (hunk 3 above)
Comment (nataliyah123) [reply tensorflow_addons/metrics/hamming.py:46]: Metric does not have any attribute result. SO, I only change the print statement.

> Thanks for the contribution! Some requests:
> 
> 1. Can we canonicalize naming convention? Like
> 
> ```python
> >>> y_true = np.foo(xxx)
> >>> y_pred = np.foo(xxx)
> >>> metric = tfa.metrics.bar(yyy)
> >>> metric.update_state(y_true, y_pred)
> >>> result = metric.result()
> >>> result.numpy()
> ```
> 
> 1. Some comments are not so meaningful in testable docstring. Like `# metric name`: When users see the API page, they know this page is that metric. Therefore, we can also clean up this kind of comments in docstring in this PR!

To the best of my knowledge, I have not encountered `# metric name`: in this Pr
